src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -420,7 +420,10 @@ class Peer(
         logger.info { "starting swap-in watch job" }
         if (swapInJob != null) return
         // wait to have a swap-in feerate available
+        logger.info { "waiting for feerates" }
         swapInFeeratesFlow.filterNotNull().first()
+        logger.info { "waiting for peer to be ready" }
+        waitForPeerReady()
```

Comment:
Moving this to the `processSwapInCommands` function, right before sending the `RequestChannelOpen`, would probably be more correct, but here it is simpler conceptually and consistent with waiting for feerates.

The rationale is to not try to reach a perfect solution, but a simple one which at worst guarantee that things fix themselves by simply restarting the app.